dbuf/dbuf/retry_forever.py

```diff
@@ -0,0 +1,19 @@
+import asyncio
+
+from .logging import log
+
+
+async def retry_forever(f, msg=None):
+    i = 0
+    while True:
+        try:
+            await f()
+            break
+        except Exception as exc:
+            if msg:
+                log.warning(msg(exc))
```

Comment:
This won't include the stack trace.  I'd prefer `log.exception(msg if msg else 'caught exception')` or something.

Reply:
I figure retry_forever shouldn't be an error level message though, and I'm pretty sure log exception is. I'll write something that grabs the stack trace and prints a warn level message.

Reply:
This would be awesome.  I think we use log.exception in a bunch of places for the exception details (good) that should just be warn or info.